README.md

```diff
@@ -4,364 +4,565 @@ The purpose of the ODM is to support wastewater-based surveillance and epidemiol
+3. `group`: This defines activities for a rule that groups or combines rules together for execution. The full rule will require the `key` and `operator` columns to be fully specified.
+4. `share`: This defines a rule that specifies the `organizationID` or `contactID` with which an output will be shared, as well as the rules to apply to generate the specific output data. The full rule will require the `key` column, but not the `operator` column, to be fully specified.
 
-This step uses two columns, `tableName` and `variableName`. The `tableName` column can specify the name(s) of the table this rules applies to and the `variableName` column can specify the name(s) of the columns this rule applies to.
+Generally, the bulk of a sharing csv will be composed of `filter` and `select` rules, with a few `group` rules, and the final `share` rules at the very end. Rules should also be written and specified in this same order.
 
-Some examples are given below:
+### 3. Selecting an Entity
+
+In order to generate an intelligible output dataset, several `select` and `filter` rules will need to first be stacked and applied. This step involves selecting the parts of the PHES-ODM or entities within the model. The entities that can be selected are:
 
-1. Selecting the `type` column in the `Sample` table
+-   Data contained in a table
+-   Data contained in column(s) of table(s)
+-   Data contained in row(s) of table(s)
 
-    | ruleId | tableName  | variableName |
-    | ------ | ---------- | ------------ |
-    | 1      | Sample     | type         |
+This step uses four columns, `table`, `mode`, `key`, and/or `value`. The `table` column specifies the name(s) of the table(s) to which this rule applies. The `mode` column specifies the action of a rule the name(s) of the column(s) to be included in the shared data output as specified by this rule. For rules that select entities, the `filter` and `select` modes will be used.
 
-2. Selecting the `qualityFlag` and the `type` column in the `WWMeasure` table
+#### 3.1. Selecting Columns
 
-    | ruleId | tableName   | variableName     |
-    | ------ | ----------- | ---------------- |
-    | 2      | WWMeasure   | qualityFlag;type |
+In order to have any data to share, tables and columns need to be specified for inclusion. These are the first rules to define in your schema. To specify which columns should be shared, specify the table or tables in the `table` column, list `select` in the `mode` column, and then list the column or columns to be shared in the `value` column. When specifying the columns, you can separate distinct column names with a ";", or if choosing several sequential columns you can list the first and last of the sequential series separated by a ":" (ex: column2:column5). The `key` and `operator` columns should be left blank (or `NA`) as they are not used in these rules, and any values in these columns for `select`-mode rows will be ignored.
```

Comment:
What defines the "sequence" when using a ":". Is it how the columns are sequenced in the data? how they are sequenced in the data dictionary?

Reply:
Maybe add that multiple tables can also be separated by a ";"

Reply:
To your first comment: both? The sequence of columns in set in the data dictionary, but should also match the sequence of columns in the data. Realistically, I guess, it should match the sequence in the data since they may omit some columns from the model. I'll add a note on this. And the semi-colon-multiple-tables-thing.